lib/ayesql.ex

```diff
@@ -356,7 +356,7 @@ defmodule AyeSQL do
   ```
   """
   defmacro defqueries(file) do
-    with {:ok, contents}  <- File.read(file),
+    with {:ok, contents}  <- File.read!(file),
          {:ok, tokens, _} <- :queries_lexer.tokenize(contents),
          {:ok, ast}       <- :queries_parser.parse(tokens) do
       functions =
```

Comment:
We could change this `with` statement body to be inside a function (mainly because the indentation would make it hard to notice that this `with` statement will have an `else` clause).
```suggestion
      gen_queries(ast)
```
where `gen_queries/1` would be a function inside this module:
```elixir
def gen_queries(ast) do
  for query <- Enum.map(ast, &AyeSQL.create_query/1) do
    quote do: unquote(query)
  end
end
```

Reply:
For testing this, we might use `assert_raise File.Error, fn -> defmodule ... end`. Not sure if this is going to work.

Reply:
Hi, thanks for the feedback! Fix the problem with the return value and added a test to ensure the exception is raised on a non-existing file.

Reply:
I still think we should not only raise when the file is not found, but also when the lexer and the parser return an error, that's why I prefer the `with` approach. Either way, for now is a good solution :) I will merge it right away!